server/src/main/java/io/crate/execution/ddl/tables/DropColumnTask.java

```diff
@@ -84,17 +87,33 @@ public ClusterState execute(ClusterState currentState, DropColumnRequest request
                                         return dc.ref();
                                     });
 
-        Metadata.Builder metadataBuilder = Metadata.builder(currentState.metadata());
-        Map<String, Object> mapping = createMappingForDroppedCols(currentTable, refsToDrop);
+        // Prepare columns tree, it will be used twice:
+        // 1. For creating mapping for dropped columns.
+        // 2. For creating "path-only, no properties" mapping to remove.
+        List<Reference> references = new ArrayList<>(refsToDrop);
+        for (var colToDrop : refsToDrop) {
+            ColumnIdent parent = colToDrop.column();
+            while ((parent = parent.getParent()) != null) {
+                if (!Symbols.containsColumn(refsToDrop, parent)
+                    && !Symbols.containsColumn(references, parent)) {
+                    references.add(currentTable.getReference(parent));
+                }
+            }
+        }
+        HashMap<ColumnIdent, List<Reference>> tree = buildTree(references);
+
+        Map<String, Object> mapping = createMappingForDroppedCols(currentTable, tree);
+        Map<String, Object> mappingToRemove = createMappingToRemove(tree, null);
 
+        Metadata.Builder metadataBuilder = Metadata.builder(currentState.metadata());
         String templateName = PartitionName.templateName(request.relationName().schema(), request.relationName().name());
         IndexTemplateMetadata indexTemplateMetadata = currentState.metadata().templates().get(templateName);
         if (indexTemplateMetadata != null) {
             // Partitioned table
             IndexTemplateMetadata newIndexTemplateMetadata = DDLClusterStateHelpers.updateTemplate(
                 indexTemplateMetadata,
                 mapping,
-                Collections.emptyMap(),
+                mappingToRemove,
```

Comment:
Template mapping code already has api (`DDLClusterStateHelpers.removeFromMapping`) to remove mappings (also used in `TransportDropConstraintAction`) made it public and slightly modified to be able to drop the whole map entries, not only map keys.